File: app/src/main/java/com/example/pantallap/ui/componente/fragmentComponente.java
package com.example.pantallap.ui.componente;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.pantallap.R;
import com.google.android.material.appbar.AppBarLayout;
import com.google.android.material.tabs.TabLayout;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link fragmentComponente#newInstance} factory method to
 * create an instance of this fragment.
 */
public class fragmentComponente extends Fragment {

    View vista;
    private AppBarLayout appbar;
    private TabLayout pestañas;
    private ViewPager viewPager;
    private PagerAdapter adapter;

    public fragmentComponente() {
        // Required empty public constructor
    }

    public static fragmentComponente newInstance(String param1, String param2) {
        fragmentComponente fragment = new fragmentComponente();
        Bundle args = new Bundle();
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        vista = inflater.inflate(R.layout.fragment_componente, container, false);

      /* pestañas = vista.findViewById(R.id.tab);
       viewPager = vista.findViewById(R.id.viewp);

       pestañas.addTab(pestañas.newTab().setIcon(R.drawable.ic_menu_camera).setText("LG"));
       pestañas.addTab(pestañas.newTab().setIcon(R.drawable.ic_celular).setText("Xiaomi"));

       adapter = new PagerAdapter(getChildFragmentManager(),pestañas.getTabCount());
       viewPager.setAdapter(adapter);
       viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(pestañas));
       pestañas.setOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
           @Override
           public void onTabSelected(TabLayout.Tab tab) {
               viewPager.setCurrentItem(tab.getPosition());
           }

           @Override
           public void onTabUnselected(TabLayout.Tab tab) {

           }

           @Override
           public void onTabReselected(TabLayout.Tab tab) {

           }
       });*/

        return vista;
    }

   /* public class PagerAdapter extends FragmentPagerAdapter{

        int numeroTab;
        public PagerAdapter(FragmentManager fm, int numerosTab) {
            super(fm);
            this.numeroTab = numerosTab;
        }
        @Override
        public Fragment getItem(int position) {
            Fragment fragment = null;
            switch (position){
                case 0: fragment = new fragmentContenedor(); break;
                case 1: fragment = new Xiaomi(); break;
            }
            return fragment;
        }

        @Override
        public int getCount() {
            return numeroTab;
        }
    }*/
}

File: app/src/main/java/com/example/pantallap/ui/componente/FragmentComponente.kt
package com.example.pantallap.ui.componente

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import com.example.pantallap.Componentes.*
import com.example.pantallap.R
import com.example.pantallap.Smartwatch.*
import com.example.pantallap.ui.Smartwatch
import com.google.android.material.tabs.TabLayout

class FragmentComponente : Fragment() {

    private lateinit var pestañas: TabLayout
    private lateinit var viewPager: ViewPager
    private lateinit var adapter: PagerAdapter

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val root = inflater.inflate(R.layout.fragment_componente, container, false)

        pestañas = root.findViewById(R.id.tab1)
        viewPager = root.findViewById(R.id.viewp1)

        pestañas.addTab(pestañas.newTab().setIcon(R.drawable.discoduro).setText("Disco Duro"))
        pestañas.addTab(pestañas.newTab().setIcon(R.drawable.memoriaram).setText("Memoria RAM"))
        pestañas.addTab(pestañas.newTab().setIcon(R.drawable.placabase).setText("Placa Base"))
        pestañas.addTab(pestañas.newTab().setIcon(R.drawable.procesador).setText("Procesadores"))
        pestañas.addTab(pestañas.newTab().setIcon(R.drawable.tarjetagrafica).setText("Tarjeta Grafica"))
        pestañas.addTab(pestañas.newTab().setIcon(R.drawable.otros).setText("Otros"))

        adapter = PagerAdapter(childFragmentManager, pestañas.getTabCount())
        viewPager.setAdapter(adapter)
        viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(pestañas))

        pestañas.tabMode = TabLayout.MODE_SCROLLABLE

        pestañas.setOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                viewPager.setCurrentItem(tab.position)
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {}
            override fun onTabReselected(tab: TabLayout.Tab) {}
        })

        return root
    }

    class PagerAdapter(fm: FragmentManager?, var numeroTab: Int) :
        FragmentPagerAdapter(fm!!) {
        override fun getItem(position: Int): Fragment {
            var fragment: Fragment? = null
            when (position) {
                0 -> fragment = DiscoDuro()
                1 -> fragment = MemoriaRAM()
                2 -> fragment = PlacasBase()
                3 -> fragment = Procesadores()
                4 -> fragment = TarjetaGrafica()
                5 -> fragment = Otros()
            }
            return fragment!!
        }

        override fun getCount(): Int {
            return numeroTab
        }
    }
}